grub.d/99_migration

```diff
@@ -21,19 +21,22 @@ root_device=$(
 )
 
 # check if location of migration_iso is not on the root_device
-image_fs_device=${root_device}
-index=4
-for path in "/usr/share" "/usr";do
-    if mountpoint -q ${path};then
-        image_fs_device=$(
-            lsblk -p -n -r -o NAME,MOUNTPOINT |\
-            grep -E "${path}$" | uniq | cut -f1 -d" "
-        )
-        migration_iso=/$(echo "${migration_iso}" | cut -f"${index}"- -d/)
-        break
-    fi
-    index=$((index -1))
-done
+if [ -f "${migration_iso}" ]; then
+    image_fs_device=${root_device}
+else
+    index=4
+    for path in "/usr/share" "/usr";do
+        if mountpoint -q ${path};then
+            image_fs_device=$(
+                lsblk -p -n -r -o NAME,MOUNTPOINT |\
+                grep -E "${path}$" | uniq | cut -f1 -d" "
+            )
+            migration_iso=/$(echo "${migration_iso}" | cut -f"${index}"- -d/)
+            break
+        fi
+        index=$((index -1))
+    done
+fi
```

Comment:
As I suggested I would drop the entire code block. With the changed target to `/migration-image` instead of `/usr/share/migration-image` none of the special sauce above should be needed anymore

That should turn the PR into a very nice state :) less code is better :)

Only if you agree of course